akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala

```diff
@@ -1353,8 +1411,26 @@ class DDataShardCoordinator(
     }
   }
 
-  def sendAllShardsUpdate(newShard: String) = {
-    replicator ! Update(AllShardsKey, GSet.empty[String], writeMajority, Some(newShard))(_ + newShard)
+  def rememberShardAllocated(newShard: String) = {
+    log.debug("Remembering shard allocation [{}]", newShard)
+    rememberEntitiesStore.foreach(_ ! RememberEntitiesCoordinatorStore.AddShard(newShard))
+    timers.startSingleTimer(
+      RememberEntitiesTimeoutKey,
+      RememberEntitiesTimeout(newShard),
+      // FIXME more reasonable timeout here?
+      settings.tuningParameters.updatingStateTimeout)
+  }
+
+  override def receiveTerminated: Receive =
+    super.receiveTerminated.orElse {
+      case RememberEntitiesStoreStopped =>
+        onRememberEntitiesStoreStopped()
+    }
+
+  def onRememberEntitiesStoreStopped(): Unit = {
+    // rely on backoff supervision of coordinator
+    log.error("The ShardCoordinator stopping because the remember entities store stopped")
+    context.stop(self)
```

Comment:
I guess that is what happens also when the coordinator state write fails when using persistent mode?

Ordinary ddata write failures doesn't end up here, they are retried?

Reply:
Yep, a bit of ahead of time implementation here since we don't have a persistent remember entitites store that you can run together with the ddata state store mode.